dhcpv4/nclient4/client.go

```diff
@@ -63,35 +74,61 @@ type pendingCh struct {
 	ch chan<- *dhcpv4.DHCPv4
 }
 
-type logger interface {
-	Printf(format string, v ...interface{})
+// Logger is a handler which will be used to output logging messages
+type Logger interface {
+	// PrintMessage print _all_ DHCP messages
 	PrintMessage(prefix string, message *dhcpv4.DHCPv4)
+
+	// Printf is use to print the rest debugging information
+	Printf(format string, v ...interface{})
 }
 
-type emptyLogger struct{}
+// EmptyLogger prints nothing
+type EmptyLogger struct{}
+
+// Printf is just a dummy function that does nothing
+func (e EmptyLogger) Printf(format string, v ...interface{}) {}
 
-func (e emptyLogger) Printf(format string, v ...interface{})             {}
-func (e emptyLogger) PrintMessage(prefix string, message *dhcpv4.DHCPv4) {}
+// PrintMessage is just a dummy function that does nothing
+func (e EmptyLogger) PrintMessage(prefix string, message *dhcpv4.DHCPv4) {}
 
-type shortSummaryLogger struct {
-	*log.Logger
+// Printfer is used for actual output of the logger. For example *log.Logger is a Printfer.
+type Printfer interface {
```

Comment:
Do we actually need this new interface?

Reply:
I use the same interface in multiple `struct`-s, so it's just DRY. And it's public, because `DebugLogger` and `ShortSummaryLogger` are public.

Reply:
Why not use `*log.Logger` inside `DebugLogger` and `ShortSummaryLogger`?

Reply:
"Require less, provide more"

We don't need exactly `*log.Logger`. We need just some `Printf`-er here (not more). Somebody else could use `logrus` or whatever else instead of `*log.Logger`. If you require only the thing you really need you make this library much more flexible. There's only one real disadvantage I know: handling through interfaces is slower than handling directly through an implementation.